src/Generator/Generators/CSharp/CSharpSources.cs

```diff
@@ -782,6 +789,12 @@ private bool CanUseSequentialLayout(Class @class)
                 {
                     if (fields[i].Offset == fields[i - 1].Offset)
                         return false;
+
+                    if (fields[i].QualifiedType.Type.Desugar().TryGetDeclaration(out Declaration declaration))
+                    {
+                        if (declaration.AlignAs != 0)
```

Comment:
Merge the conditions.

Reply:
I did that to avoid a long line but I've created a variable for the "desugared" type and merged both conditions.